temporalio/lib/temporalio/internal/worker/workflow_instance.rb

```diff
@@ -626,9 +649,11 @@ def on_top_level_exception(err)
         end
 
         def failure_exception?(err)
-          err.is_a?(Error::Failure) || err.is_a?(Timeout::Error) || @failure_exception_types.any? do |cls|
-            err.is_a?(cls)
-          end
+          err.is_a?(Error::Failure) || err.is_a?(Timeout::Error) ||
+            ((@workflow_failure_exception_types || []) +
+             (@definition_options.failure_exception_types || [])).any? do |cls|
+              err.is_a?(cls)
+            end
```

Comment:
```suggestion
            @workflow_failure_exception_types&.any? { |cls| err.is_a?(cls) } ||
            @definition_options.failure_exception_types&.any? { |cls| err.is_a?(cls) }
              err.is_a?(cls)
            end
```

Reads a bit easier to me. Sure rubocop may change to blocks or whatever, but concatenating and creating empty arrays when not needed seems unnecessary.